### PR TITLE
Make deployable as helm charts, separating monitoring from continuous-load

### PR DESCRIPTION
**Pull request by robertgmoss**
This add the components into helm charts and aggregates them into two distinct charts: the first is optional, a chart to deploy the monitoring stack; the second deploys the continuous load apps, k6 and podinfo.  

There are a number of changes that simplify the architecture:

- Istio and Nginx Ingress are no longer deployed
- The deployment of podinfo for source no longer uses a modified version and instead take from the upsteam helm chart
- The [Prometheus operator](https://prometheus-operator.dev/) is installed directly instead of with the kube-prometheus-stack helm chart, due to imcompatability with kubernetes 1.25  
- The Grafana dashboard has been updated to remove panels that reference Istio and Nginx

### CODE REVIEW

**Comment by tmcalves** [comment]
This is a big rework, can you update the readme as well? Feel very outdated now, should have some clear instructions on how to execute this and what it contains.

**Comment by robertgmoss** [comment]
@tmcalves Can you take another look please?

File: README.md [hunk 1]
```diff
@@ -1,66 +1,60 @@
...
-cd podinfo ; make integration prometheus_url=$prometheus_url source_url=$source_url target_url=$target_url region=$region cluster=$cluster
+./deploy.sh
 ```
 
-### Improvements
 
-- [Chaos testing] Use a controller like [pod-reaper](https://github.com/target/pod-reaper) to graceful shutdown target podinfo and nginx pods to exercise graceful shutdown/rolling deployment
+## Improvements
+
+- [Chaos testing] Use a controller like [pod-reaper](https://github.com/target/pod-reaper) to graceful shutdown target podinfo  pods to exercise graceful shutdown/rolling deployment
```
Comment (tmcalves) [inline README.md]: This can probably be removed since podinfo was deleted. If we want to make impvoements they can be raised as issues

Comment (robertgmoss) [reply README.md]: done

**Review by tmcalves — APPROVED** [review]
lgtm!

File: deploy.sh [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  echo "-----------------------------------------"
+  echo "-----------------------------------------" 
+}
+
+while getopts 'i' flag; do
+  case "${flag}" in
+    i)  
+      source ./prerequisites.sh
+      namespace="continuous-load"
+      create_namespace $namespace
```
Comment (chbatey) [inline deploy.sh]: could we use helm's support for creating the namespace?

**Comment by chbatey** [comment]
When i try this i get so 

```
===> Deploying Continuous Load
Error: no repository definition for https://stefanprodan.github.io/podinfo. Please add the missing repos via 'helm repo add'
```

File: deploy.sh [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+# in continuous-load/charts/k6/load.js.
+function deploy_continuous_load() {
+  local namespace="$1"
+  local loadTargetService="$2"
+  local replicas="$3"
+  local thresholds="$4"
+  local reqPerSecond="$5"
+  echo "===> Deploying Continuous Load"
+  helm dependency build ./continuous-load
+  helm upgrade -install --wait continuous-load \
```
Comment (chbatey) [inline deploy.sh:27]: can we make it do the helm chart just works without setting any parameters, so sensible default values, as a user of this isn't likely to take the shell script but integrate the helm chart into their existing path to production. So:

* A default namespace e.g. continuous-load
* A default for replicas, thresholds and rps

File: deploy.sh [hunk 4]
```diff
@@ -0,0 +1,90 @@
+#!/bin/bash -eu
+
+# deploy_continuous_load - deploys the load injector and target application as well as
+# the ServiceMonitor and PodMonitor objects that are used to supply Prometheus with
```
Comment (chbatey) [inline deploy.sh:4]: great docs, can we replicate / move most of it to the documentation of the helm chart, as i think that'll be the user interface

File: README.md [hunk 5]
```diff
@@ -1,66 +1,55 @@
...
+PrometheusRule resources are created to configure alerts on prometheus.
+Such alerts follow the [Golden Signal](https://sysdig.com/blog/golden-signals-kubernetes/) principles. 
 
 ### Dashboard 
 
-Continuous dashboard is create automatically and the link will be available within the console after running the script below.
+A GrafanaDashboard resource is used to create the continuous load dashboard.
 
-### Run 
+## Run
```
Comment (chbatey) [inline README.md]: Can we change this section to "Running Locally"

Then add a new section about real life deployment e.g.


## Deployment

There are two helm charts:

* Prerequsites: everything needed to display the dashboards for Continuous Load
* Continuous Load: The run time components of Continuous Load

For an end to end demo, install both, for production deployments, consider integrating with an existing prometheus and grafana installation.


## Pre-configured monitoring stack (pre-requsites)

TODO show how to use the helm chart directly

## Continuous Load (no dashboarding)

TODO show how to use the helm chart directly

## Deploying dashboards

TODO show how to use an existing ns

File: README.md [hunk 6]
```diff
@@ -1,66 +1,55 @@
-## Continuous Load
+# Continuous Load
 
-Continuous Load is a project design to run load 24/7 across an infrastructure and monitor the network health even in production. 
-We have rolled out various versions at clients.
+Continuous Load is a project designed to run load 24/7 across an infrastructure and monitor the network health even in production.
```
Comment (chbatey) [inline README.md]: One extra like to say the high level goal i.e. "
It's goal is to find problems in infrastructure before the users of that infrastructure find them."

And can we also include an image of the dashboard?

File: README.md [hunk 7]
```diff
@@ -1,66 +1,55 @@
...
-
-Integration tests have been created [here](podinfo/test/integration/integration_test.go) but will not work for the local version, it needs a bit of work.
+# delete the unnecessary deployment
+kubectl -n <namespace> delete deployment.apps/prometheus-operator
+```
 
+To install only the continuous-load components run without the flag:
 ```
-cd podinfo ; make integration prometheus_url=$prometheus_url source_url=$source_url target_url=$target_url region=$region cluster=$cluster
+./deploy.sh
```
Comment (chbatey) [inline README.md:56]: add a section on what i do next to see the dashboard, assume this is the first time someone has tried to use this

**Comment by chbatey** [comment]
I followed the instructions but can't see the dashboard (i did deploy -i)

Any more things i need to do?

image

**Comment by robertgmoss** [comment]
That should be fixed now - it adds extra field to the json when exporting and these break the dashboard.  I committed these by mistake.

File: continuous-load-dashboard.yaml [hunk 8]
```diff
@@ -0,0 +1,1457 @@
+---
+apiVersion: grafana.integreatly.org/v1beta1
+kind: GrafanaDashboard
+metadata:
+  name: grafanadashboard-sample
```
Comment (chbatey) [inline continuous-load-dashboard.yaml]: can we call it continuous-load?

File: monitoring/charts/grafana/templates/grafana.yaml [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+  labels:
+    dashboards: "grafana"
+spec:
+  config:
+    log:
+      mode: "console"
+    auth:
+      disable_login_form: "false"
+    security:
+      admin_user: root
```
Comment (chbatey) [inline monitoring/charts/grafana/templates/grafana.yaml]: if we remove these will they be the default of admin/admin? Probably better so the user doesn't need to go find it and they are equally insecure

**Review by chbatey — COMMENTED** [review]
All worked for me this time, nice!

Two last comments then LGTM

File: monitoring/charts/grafana/templates/grafana.yaml (hunk 9 above)
Comment (robertgmoss) [reply monitoring/charts/grafana/templates/grafana.yaml]: Leaving it out doesn't work.  Setting it to admin/admin causes it to prompt the user to change the password immediately, which in turn means the `deploy.sh` script can't deploy the dashboard successfully.

I can add the details to the readme.

Comment (chbatey) [reply monitoring/charts/grafana/templates/grafana.yaml]: aha! gotcha, let's just add it to the readme then

**Pull request merged**